enterprise/frontend/src/embedding-sdk/components/private/QuestionEditor/QuestionEditor.tsx

```diff
@@ -0,0 +1,82 @@
+import { useDisclosure } from "@mantine/hooks";
+import { useState } from "react";
+
+import type { InteractiveQuestionProps } from "embedding-sdk/components/public/InteractiveQuestion";
+import { InteractiveQuestion } from "embedding-sdk/components/public/InteractiveQuestion";
+import { Box, Group, Overlay, Paper, Tabs } from "metabase/ui";
+
+import { useInteractiveQuestionContext } from "../InteractiveQuestion/context";
+
+const QuestionEditorInner = () => {
+  const { queryResults, runQuestion } = useInteractiveQuestionContext();
+
+  const [activeTab, setActiveTab] = useState<
+    "notebook" | "visualization" | (string & unknown) | null
```

Comment:
What is the `(string & unknown)` part of the type for? Since I see we the have `notebook` and `visualization` tab here

Reply:
ah that's to make the Mantine `Tab` component happy. Mantine expects a string for the activeTab prop but I wanted to make sure that we could have some level of type narrowing for my own (and hopefully other dev)'s happiness, so I had to do this. [This slack thread explains it better than I could](https://metaboat.slack.com/archives/C505ZNNH4/p1723030485597639)